config/unicorn.conf.rb

```diff
@@ -89,13 +89,25 @@
       Demon::Sidekiq.after_fork { DiscourseEvent.trigger(:sidekiq_fork_started) }
       Demon::Sidekiq.start(sidekiqs, logger: server.logger)
 
-      # Trap USR1, so we can re-issue to sidekiq workers
-      # but chain the default unicorn implementation as well
-      old_handler =
-        Signal.trap("USR1") do
-          Demon::Sidekiq.kill("USR1")
-          old_handler.call
-        end
+      if Discourse.enable_sidekiq_logging?
+        # Trap USR1, so we can re-issue to sidekiq workers
+        # but chain the default unicorn implementation as well
+        old_handler =
+          Signal.trap("USR1") do
+            old_handler.call
+
+            Thread.new do
```

Comment:
Should we eventually `join` this thread? I do not know what happens internally, but I am worrying about dangling terminated threads.

Reply:
Good catch! I should run the old_handler.call in the thread instead and have it sleep 1 there.

Reply:
I figured out another way to implement the behaviour without using any threads. Thank you for catching this @nbianca 👍